monitoring/perf_context.cc

```diff
@@ -37,6 +37,181 @@ PerfContext::~PerfContext() {
+    seek_min_heap_time = other.seek_min_heap_time;
+    seek_internal_seek_time = other.seek_internal_seek_time;
+    find_next_user_entry_time = other.find_next_user_entry_time;
+    write_pre_and_post_process_time = other.write_pre_and_post_process_time;
+    write_memtable_time = other.write_memtable_time;
+    write_delay_time = other.write_delay_time;
+    write_thread_wait_nanos = other.write_thread_wait_nanos;
+    write_scheduling_flushes_compactions_time = other.write_scheduling_flushes_compactions_time;
+    db_mutex_lock_nanos = other.db_mutex_lock_nanos;
+    db_condition_wait_nanos = other.db_condition_wait_nanos;
+    merge_operator_time_nanos = other.merge_operator_time_nanos;
+    read_index_block_nanos = other.read_index_block_nanos;
+    read_filter_block_nanos = other.read_filter_block_nanos;
+    new_table_block_iter_nanos = other.new_table_block_iter_nanos;
+    new_table_iterator_nanos = other.new_table_iterator_nanos;
+    block_seek_nanos = other.block_seek_nanos;
+    find_table_nanos = other.find_table_nanos;
+    bloom_memtable_hit_count = other.bloom_memtable_hit_count;
+    bloom_memtable_miss_count = other.bloom_memtable_miss_count;
+    bloom_sst_hit_count = other.bloom_sst_hit_count;
+    bloom_sst_miss_count = other.bloom_sst_miss_count;
+    key_lock_wait_time = other.key_lock_wait_time;
+    key_lock_wait_count = other.key_lock_wait_count;
+
+    env_new_sequential_file_nanos = other.env_new_sequential_file_nanos;
+    env_new_random_access_file_nanos = other.env_new_random_access_file_nanos;
+    env_new_writable_file_nanos = other.env_new_writable_file_nanos;
+    env_reuse_writable_file_nanos = other.env_reuse_writable_file_nanos;
+    env_new_random_rw_file_nanos = other.env_new_random_rw_file_nanos;
+    env_new_directory_nanos = other.env_new_directory_nanos;
+    env_file_exists_nanos = other.env_file_exists_nanos;
+    env_get_children_nanos = other.env_get_children_nanos;
+    env_get_children_file_attributes_nanos = other.env_get_children_file_attributes_nanos;
+    env_delete_file_nanos = other.env_delete_file_nanos;
+    env_create_dir_nanos = other.env_create_dir_nanos;
+    env_create_dir_if_missing_nanos = other.env_create_dir_if_missing_nanos;
+    env_delete_dir_nanos = other.env_delete_dir_nanos;
+    env_get_file_size_nanos = other.env_get_file_size_nanos;
+    env_get_file_modification_time_nanos = other.env_get_file_modification_time_nanos;
+    env_rename_file_nanos = other.env_rename_file_nanos;
+    env_link_file_nanos = other.env_link_file_nanos;
+    env_lock_file_nanos = other.env_lock_file_nanos;
+    env_unlock_file_nanos = other.env_unlock_file_nanos;
+    env_new_logger_nanos = other.env_new_logger_nanos;
+    get_cpu_nanos = other.get_cpu_nanos;
+    if (other.per_level_perf_context_enabled &&
+        other.level_to_perf_context != nullptr) {
+      ClearPerLevelPerfContext();
+      EnablePerLevelPerfContext();
+      *level_to_perf_context = *other.level_to_perf_context;
```

Comment:
This needs to be copied, no matter `other.per_level_perf_context_enabled` is true or false.

Reply:
you mean create an iterator and for loop kind of copy? I thought std::map::operator= works fine in this case? https://en.cppreference.com/w/cpp/container/map/operator%3D

Reply:
This is not what I mean. I mean we need to remove the `other.per_level_perf_context_enabled` check in if.

Reply:
oh, forgive my poor understanding, somehow I totally missed the second part